clubs/mixins.py

```diff
@@ -0,0 +1,241 @@
+
+    Changes the default filename to include the date and time of creation.
+    Changes the default column header to be bolded.
+    """
+
+    def _format_header_value(self, key):
+        """
+        Format the text displayed in the column header.
+        """
+        return key.replace('_', ' ').title()
+
+    def _many_to_many_individual_formatter(self, value):
+        """
+        Take in an individual object generated by a ManyToMany serializer and format it.
+        If there is a single field in the dictionary, return that. Otherwise, return the
+        first field that is not the id field.
+        """
+        if isinstance(value, dict):
+            if len(value) == 1:
+                return next(value.values())
+            elif len(value) > 1:
+                return next(v for k, v in value.items() if not k == 'id')
+        return str(value)
+
+    def _many_to_many_formatter(self, value):
+        """
+        Take in output generated by a ManyToMany serializer and format it.
+        """
+        if isinstance(value, list):
+            return ', '.join(self._many_to_many_individual_formatter(v) for v in value)
+        return value
+
+    def _lookup_field_formatter(self, field):
+        """
+        Return a method that can format the field given the name of the field.
+        """
+        # if there is no model serializer, don't format
+        try:
+            serializer = self.get_serializer_class()
+        except AttributeError:
+            return lambda x: x
+
+        if not isinstance(serializer, serializers.SerializerMetaclass):
+            return lambda x: x
+
+        # allow serializer to override field formatting
+        key = 'format_{}_for_spreadsheet'.format(field)
+        if hasattr(serializer, key):
+            return lambda x: getattr(serializer, key)(serializer, x)
+
+        # lookup column type from serializer
+        if field in serializer._declared_fields:
+            serializer_field_object = serializer._declared_fields[field]
+            if isinstance(serializer_field_object, serializers.SerializerMethodField):
+                return lambda x: x
+
+        # lookup column type from model
+        model = serializer.Meta.model
+        try:
+            field_object = model._meta.get_field(field)
+        except FieldDoesNotExist:
+            # if model field lookup fails, rely on serializer field
+            if isinstance(serializer_field_object, serializers.BooleanField):
+                return lambda x: str(bool(x))
+            return lambda x: x
+
+        # format based on field type
+        if isinstance(field_object, ManyToManyField):
+            return self._many_to_many_formatter
+        elif isinstance(field_object, BooleanField):
+            return lambda x: str(bool(x))
+        elif hasattr(field_object, 'choices') and field_object.choices is not None:
+            choices = dict(field_object.choices)
+            return lambda x: choices.get(x, 'Unknown')
```

Comment:
Would it make sense to just return `x` in the case it's not in the choices dict? might not be in the right format but i feel like it would give more information than `Unknown` would.

Reply:
Ideally you shouldn't even be able to save something that's not in the choices dict. I thought that this would be less confusing than the alternative, since the alternative would usually be some random number (in the case of an `IntegerField`) or meaningless sequence of letters (in the case of an `CharField`). Let me know what you think!

Reply:
Makes sense to me :)